File: 2085-count-common-words-with-one-occurrence/2085-count-common-words-with-one-occurrence.cpp
class Solution {
public:
    int countWords(vector<string>& words1, vector<string>& words2) {
        unordered_map<string,int>mp1;
        unordered_map<string,int>mp2;
        int count=0;
        int n1=words1.size();
        int n2=words2.size();
        for(int i=0;i<n1;i++){
            mp1[words1[i]]++;
        }
        for(int i=0;i<n2;i++){
            mp2[words2[i]]++;
        }
        for(auto it:mp1){
            if(it.second==1 && mp2[it.first]==1){
                 count++;
            }
        }
        return count;
    }
};

File: 2085-count-common-words-with-one-occurrence/2085-count-common-words-with-one-occurrence.java
class Solution {
    public int countWords(String[] words1, String[] words2) {
        HashMap<String,Integer>mp1=new HashMap<>();
        HashMap<String,Integer>mp2=new HashMap<>();
        int n1=words1.length;
        int n2=words2.length;
        int count=0;
        for(int i=0;i<n1;i++){
            mp1.put(words1[i],mp1.getOrDefault(words1[i],0)+1);
        }
        for(int i=0;i<n2;i++){
            mp2.put(words2[i],mp2.getOrDefault(words2[i],0)+1);
        }
        for(Map.Entry<String,Integer>entry: mp1.entrySet()){
              if(entry.getValue()==1 && mp2.getOrDefault(entry.getKey(),0)==1){
                 count++;
              }
        }
        return count;
    }
}